server/follower_controller.go

```diff
@@ -267,16 +267,6 @@ func (fc *followerController) NewTerm(req *proto.NewTermRequest) (*proto.NewTerm
 			slog.Int64("new-term", req.Term),
 		)
 		return nil, common.ErrorInvalidTerm
-	} else if req.Term == fc.term && fc.status != proto.ServingStatus_FENCED {
```

Comment:
If they are same term, how about return the `lastEntryId, err := getLastEntryIdInWal(fc.wal)` directly?